gnomad_hail/utils/generic.py

```diff
@@ -56,6 +56,72 @@ def unphase_mt(mt: hl.MatrixTable) -> hl.MatrixTable:
                                )
 
 
+def get_reference_ht(
+        ref: hl.ReferenceGenome,
+        contigs: Optional[List[str]] = None,
+        excluded_intervals: Optional[List[hl.Interval]] = None,
+        add_all_substitutions: bool = False,
+        filter_n: bool = True
+) -> hl.Table:
+    """
+    Creates a reference Table with locus and alleles (containing only the reference allele by default) from the given reference genome.
+
+    .. note::
+
+        If the `contigs` argument is not provided, all contigs (including obscure ones) will be added to the table.
+        This can be slow as contigs are added one by one.
+
+    :param ref: Input reference genome
+    :param contigs: An optional list of contigs that the Table should include
+    :param excluded_intervals: An optional list of intervals to exclude
+    :param add_all_substitutions: If set, then all possible substitutions are added in the alleles array
+    :param filter_n: If set, bases where the reference is unknown (n) are filtered.
+    :return:
+    """
+    if not ref.has_sequence():
+        add_reference_sequence(ref)
+
+    if not contigs:
+        contigs = ref.contigs
+
+    if add_all_substitutions:
+        SUBSTITUTIONS_TABLE = hl.literal({
+            'a': ['c', 'g', 't'],
+            'c': ['a', 'g', 't'],
+            'g': ['a', 'c', 't'],
+            't': ['a', 'c', 'g']
+        })
+
+    context = None
+    for contig in contigs:
+        _context = hl.utils.range_table(ref.contig_length(contig), n_partitions=int(ref.contig_length(contig) / 5000000))
```

Comment:
no specific reason really..seemed reasonable given that for each row there's very little data. But happy to reconsider if you have another suggestion!

Reply:
makes sense. what about adding an `n_partitions` arg that is set to 50 by default, since `len(chromosome 1) / 5000000` should be ~50? no strong feelings about this though.